core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -1748,16 +1748,20 @@ private RecordCursorFactory generateJoins(QueryModel model, SqlExecutionContext
             if (constFilter != null) {
                 Function function = functionParser.parseFunction(constFilter, null, executionContext);
                 function.init(null, executionContext);
-                if (!function.getBool(null)) {
-                    // do not copy metadata here
-                    // this would have been JoinRecordMetadata, which is new instance anyway
-                    // we have to make sure that this metadata is safely transitioned
-                    // to empty cursor factory
-                    JoinRecordMetadata metadata = (JoinRecordMetadata) master.getMetadata();
-                    metadata.incrementRefCount();
-                    RecordCursorFactory factory = new EmptyTableRecordCursorFactory(metadata);
-                    Misc.free(master);
-                    return factory;
+                try {
+                    if (!function.getBool(null)) {
+                        // do not copy metadata here
+                        // this would have been JoinRecordMetadata, which is new instance anyway
+                        // we have to make sure that this metadata is safely transitioned
+                        // to empty cursor factory
+                        JoinRecordMetadata metadata = (JoinRecordMetadata) master.getMetadata();
+                        metadata.incrementRefCount();
+                        RecordCursorFactory factory = new EmptyTableRecordCursorFactory(metadata);
+                        Misc.free(master);
+                        return factory;
+                    }
+                } catch (UnsupportedOperationException e) {
```

Comment:
we are not meant to catch these exceptions, there should be a better way to handle this

Reply:
Good idea. Changed to a type check in c445e862d6f67c77500ddc2f64e9f902111060b3